ui/app/ducks/gas/gas.duck.js

```diff
@@ -416,114 +420,111 @@ export function fetchGasEstimates(blockTime) {
+      )
+
+      const withSupplementalTimeEstimates = flatten(
+        estimatedTimeWithUniquePrices.map(
+          ({ expectedWait, gasprice }, i, arr) => {
+            const next = arr[i + 1]
+            if (!next) {
+              return [{ expectedWait, gasprice }]
+            }
+            const supplementalPrice = getRandomArbitrary(
+              gasprice,
+              next.gasprice,
+            )
+            const supplementalTime = extrapolateY({
+              higherY: next.expectedWait,
+              lowerY: expectedWait,
+              higherX: next.gasprice,
+              lowerX: gasprice,
+              xForExtrapolation: supplementalPrice,
             })
-        : Promise.resolve(
-            priceAndTimeEstimates.length
-              ? priceAndTimeEstimates
-              : loadLocalStorageData('GAS_API_ESTIMATES'),
-          )
-
-    return promiseToFetch.then((estimates) => {
-      dispatch(setPricesAndTimeEstimates(estimates))
-      dispatch(gasEstimatesLoadingFinished())
-    })
+            const supplementalPrice2 = getRandomArbitrary(
+              supplementalPrice,
+              next.gasprice,
+            )
+            const supplementalTime2 = extrapolateY({
+              higherY: next.expectedWait,
+              lowerY: supplementalTime,
+              higherX: next.gasprice,
+              lowerX: supplementalPrice,
+              xForExtrapolation: supplementalPrice2,
+            })
+            return [
+              { expectedWait, gasprice },
+              {
+                expectedWait: supplementalTime,
+                gasprice: supplementalPrice,
+              },
+              {
+                expectedWait: supplementalTime2,
+                gasprice: supplementalPrice2,
+              },
+            ]
+          },
+        ),
+      )
+      const withOutliersRemoved = inliersByIQR(
+        withSupplementalTimeEstimates.slice(0).reverse(),
+        'expectedWait',
+      ).reverse()
+      const timeMappedToSeconds = withOutliersRemoved.map(
+        ({ expectedWait, gasprice }) => {
+          const expectedTime = new BigNumber(expectedWait)
+            .times(Number(blockTime), 10)
+            .toNumber()
+          return {
+            expectedTime,
+            gasprice: new BigNumber(gasprice, 10).toNumber(),
+          }
+        },
+      )
+
+      const timeRetrieved = Date.now()
+      dispatch(setApiEstimatesLastRetrieved(timeRetrieved))
+
+      await Promise.all([
+        setStorageItem('GAS_API_ESTIMATES_LAST_RETRIEVED', timeRetrieved),
+        setStorageItem('GAS_API_ESTIMATES', timeMappedToSeconds),
+      ])
+
+      estimates = timeMappedToSeconds
+    } else if (priceAndTimeEstimates.length) {
+      estimates = priceAndTimeEstimates
+    }
+
+    dispatch(setPricesAndTimeEstimates(estimates))
+    dispatch(gasEstimatesLoadingFinished())
+    return estimates
   }
 }
 
 export function setCustomGasPriceForRetry(newPrice) {
```

Comment:
This is being called in `useRetryTransaction` in an async context, but without using `await`. Now that it's async, I think that hook should block until this is finished, so that the sidebar isn't shown before this completes.